crates/datasources/src/lance/mod.rs

```diff
@@ -117,7 +117,7 @@ impl ExecutionPlan for LanceInsertExecPlan {
     }
 
     fn schema(&self) -> SchemaRef {
-        TableProvider::schema(&self.dataset)
+        COUNT_SCHEMA.clone()
```

Comment:
Is this unrelated to flightsql?

Reply:
yeah, I can pull this into a separate PR, mostly didn't want to lose track of this. 

I think our exec plans in general say that they have the schema of their _inputs_ and not their _outputs_ so the problem is not localized to this case.